roboflow/models/video.py

```diff
@@ -0,0 +1,228 @@
+import json
+import time
+from typing import List
+from urllib.parse import urljoin
+
+import magic
+import requests
+
+from roboflow.config import API_URL
+from roboflow.models.inference import InferenceModel
+
+SUPPORTED_ROBOFLOW_MODELS = ["object-detection", "classification", "instance-segmentation"]
+
+SUPPORTED_ADDITIONAL_MODELS = {
+    "clip": {
+        "model_id": "clip",
+        "model_version": "1",
+        "inference_type": "clip-embed-image"
+    },
+    "gaze": {
+        "model_id": "gaze",
+        "model_version": "1",
+        "inference_type": "gaze-detection"
+    }
+}
+
+
+def is_valid_mime(filename):
+    mime = magic.Magic(mime=True)
+    file_type = mime.from_file(filename)
+    return file_type in ["video/mp4", "video/avi", "video/webm"]
+
+
+def is_valid_video(filename):
+    # check file type
+    if not is_valid_mime(filename):
+        return False
+    
+    return True
+
+class VideoInferenceModel(InferenceModel):
+    """
+    Run inference on an object detection model hosted on Roboflow or served through Roboflow Inference.
+    """
+
+    def __init__(
+        self,
+        api_key,
+    ):
+        """
+        Create a VideoDetectionModel object through which you can run inference on videos.
+
+        Args:
+            api_key (str): Your API key (obtained via your workspace API settings page).
+        """
+        self.__api_key = api_key
+
+    def predict(
+        self,
+        video_path: str,
+        inference_type: str,
+        fps: int = 5,
+        additional_models: list = None,
+    ) -> List[str, str]:
+        """
+        Infers detections based on image from specified model and image path.
+
+        Args:
+            video_path (str): path to the video you'd like to perform prediction on
+            inference_type (str): type of the model to run
+            fps (int): frames per second to run inference
+
+        Returns:
+            A list of the signed url and job id
+
+        Example:
+            >>> import roboflow
+
+            >>> rf = roboflow.Roboflow(api_key="")
+
+            >>> project = rf.workspace().project("PROJECT_ID")
+
+            >>> model = project.version("1").model
+
+            >>> prediction = model.predict("video.mp4", fps=5, inference_type="object-detection"
+        """
+
+        url = urljoin(API_URL, "/video_upload_signed_url/?api_key=", self.__api_key)
+
+        if fps > 5:
+            raise Exception("FPS must be less than or equal to 5.")
```

Comment:
The user needs to know that the FPS can be upto the video frame rate, but given the fps-based pricing, set this value to as low as makes sense to their app. Maybe make it part of the blogpost?

Reply:
It would be great if the server raised an exception if FPS is higher. Otherwise, if the user provides a video URL we have to download it in the package to check its FPS before we can do anything.